segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/WriteOperation.java

```diff
@@ -105,46 +104,44 @@ public CompletableFuture<Void> call() {
                 chunkedSegmentStorage.getLogPrefix(), System.identityHashCode(this), handle.getSegmentName(), offset, length);
 
         val streamSegmentName = handle.getSegmentName();
-        return ChunkedSegmentStorage.tryWith(chunkedSegmentStorage.getMetadataStore().beginTransaction(false, handle.getSegmentName()),
```

Comment:
These changes seem to be unrelated with delayed future cancelation..

Reply:
Reverted